### PR TITLE
Fill in the copyright info in the license template

### PR DESCRIPTION
**Pull request by autarch**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright 2024 MongoDB, Inc.
```
Comment (prestonvasquez) [inline LICENSE:189]: We apply the explicit copyright on a per-file basis. It's unclear to me whether we intentionally leave this line blank. AFAICT no other drivers explicitly apply a copyright. @blink1073 do you have insight?

Comment (blink1073) [reply LICENSE:189]: Correct.  This is the appendix, and is meant to be written as-is.  PyMongo and Java have the same appendix.

**Comment by prestonvasquez** [comment]
@autarch Thank you for the PR, closing for now as it appears we are compliant per [this](https://github.com/mongodb/mongo-go-driver/pull/1638#discussion_r1598673329) thread.

File: LICENSE (hunk 1 above)
Comment (autarch) [reply LICENSE:189]: Ah, I missed that this was the appendix.

**Pull request closed without merging**